lib/services/user.dart

```diff
@@ -205,12 +207,24 @@ class UserService {
     await _removeStoredFirstPostsData();
     await _removeStoredTopPostsData();
     await DiskCache().clear();
+    await clearTemporaryDirectory();
     Post.clearCache();
     User.clearNavigationCache();
     PostComment.clearCache();
     Community.clearCache();
   }
 
+  Future<bool> clearTemporaryDirectory() async {
+      try {
+        Directory tempDir = Directory((await getApplicationDocumentsDirectory()).path.replaceFirst('Documents', 'tmp'));
```

Comment:
Temporary media files (before and after compression) currently aren't stored in the `tmp` directory. Maybe these should be moved there instead so they are also cleared by this method?

Would require changes in two places: 
1) `MediaService._getTempPath()`
2) `MainActivity.createTemporaryFile()`

Reply:
In the documentation for the `getApplicationDocumentsDirectory` in the `path_provider.dart` file 
```
/// Path to a directory where the application may place data that is
/// user-generated, or that cannot otherwise be recreated by your application.
///
/// On iOS, this uses the `NSDocumentDirectory` API. Consider using
/// [getApplicationSupportDirectory] instead if the data is not user-generated.
///
/// On Android, this uses the `getDataDirectory` API on the context. Consider
/// using [getExternalStorageDirectory] instead if data is intended to be visible
/// to the user.
Future<Directory> getApplicationDocumentsDirectory() async {
... 
...
```
So flutter seems to suggest that user-generated data should be placed in documents. Another thing to note is, I make sure I delete respective uncompressed, compressed and thumbnail files during post creation for all user paths. So there is nothing left in the documents directory to clear.

Reply:
Another weird thing i noticed was, that the `getTemporaryDirectory` method actually returns the path to documents directory, not the `/tmp` directory.. which is why im doing a string replace. I wonder if this is a bug.

Reply:
While these files are generated by a user action I wouldn't call them user-generated. Not like the notes you write in your memo app, or the recipes you have in your cooking app. They're temporary files in every sense of the word.

And yes, you clear them when the create post dialog is exited (whether it is cancelled or you go through and post). But what if the user never leaves the dialog, as far as the app is aware? A crash, swiping the app out of existence, or force-closing it will all mean that these files are left behind. While it won't happen often, and probably won't lead to much wasted space, it's still a "storage leak".

You can clear them using the "Clear Data" button (at least on Android), but that assumes there is no actual user-generated data stored there as well (like offline post drafts or something).

Reply:
> Another weird thing i noticed was, that the `getTemporaryDirectory` method actually returns the path to documents directory, not the `/tmp` directory.. which is why im doing a string replace. I wonder if this is a bug.

I checked the source code for `getTemporaryDirectory()` and it returns the correct directory at least on Android. 🤔

Reply:
Yea i was interpreting it as files generated as a result of user actions, but anyway, perhaps it does make sense to move this to the tmp directory. I dunno whats going on with ios and `getTemporaryDirectory()` though 🤔

Reply:
Could it be that on development getTemporaryDirectory is the app directory but on production is not?

Reply:
They labeled it as an documentation issue, and changed it to say the `getTemporaryDirectory` actually uses `NSCacheDirectory` instead of the `NSTemporaryDirectory` 🤷‍♂ 
https://github.com/flutter/flutter/issues/10431

Reply:
I inspected the data containers of android and ios. On ios, the `getTemporaryDirectory` exists inside a caches directory which has other flutter stuff so I cannot clear it clean without side-effects. So ive created another directory `mediaCache` inside the path returned by `getTemporaryDirectory` on both platforms and am clearing that with all the other caches. 

@Komposten though I couldnt really wrap my head around the `this.getExternalFilesDir(Environment.DIRECTORY_PICTURES))` in the `MainActivity.java` , its not clear which method is to be used to get to the cache dir. My IDE for some reason gives me no intellisense, and I cant inspect classes either 🤔  Can you help me with what should we replace that with.? 🙏

Reply:
@uiboy 
Android has two types of "internal" directories:
- `getFilesDir()`: For storing files that are private to the app and the user shouldn't be able to see from a file explorer. Basically what Flutter would refer to as "user-generated files".
-`getCacheDir()`: A cache directory. For storing files temporarily.

Flutter's `getTemporaryDirectory()` returns `getCacheDir()`, so to be consistent you would probably want to do something like replacing `getExternalFilesDir(...)` with `new File(getCacheDir(), "mediaCache")`.

IntelliSense doesn't work in the Android files for me either. It used to work in Android Studio, but now that seems broken as well...

Reply:
Cool, done!